modules/core/src/minmax.dispatch.cpp

```diff
@@ -457,10 +457,10 @@ static void reduceMinMax(cv::InputArray src, cv::OutputArray dst, ReduceMode mod
     CV_INSTRUMENT_REGION();
 
     cv::Mat srcMat = src.getMat();
-    axis = (axis + srcMat.dims) % srcMat.dims;
-    CV_Assert(srcMat.channels() == 1 && axis >= 0 && axis < srcMat.dims);
 
-    std::vector<int> sizes(srcMat.dims);
+    axis = (srcMat.dims == 0) ? 0 : (axis + srcMat.dims) % srcMat.dims;
+    CV_Assert(srcMat.channels() == 1 && axis >= 0 && axis <= srcMat.dims);
+    std::vector<int> sizes((srcMat.dims == 0) ? 1 : srcMat.dims);
```

Comment:
That looks simpler:
```cpp
int dims = max(1, srcMat.dims);
axis = (axis + dims) % dims;
CV_Assert(srcMat.channels() == 1 && axis >= 0 && axis < dims);
std::vector<int> sizes(dims);
```

Reply:
```C++
axis = (axis + dims) % dims;
```

fails on Linux. Had to change it to 

```c++
axis = (srcMat.dims == 0) ? 0 : (axis + srcMat.dims) % srcMat.dims;
```

This change it need for supporting 0D matrices: 

```
(srcMat.dims == 0) ? 1 : srcMat.dims
```

Reply:
Even with dims=1 after `int dims = max(1, srcMat.dims);`?

Reply:
oh, did not pay attention to `max`. I will check it